fleetspeak/src/server/comms.go

```diff
@@ -67,6 +67,7 @@ func (c commsContext) InitializeConnection(ctx context.Context, addr net.Addr, k
 		ClientLabels: wcd.ClientLabels,
 	}
 	if !c.s.authorizer.Allow2(addr, contactInfo) {
+		log.Warningf("Allow2 failed for %v", addr)
 		return nil, nil, false, comms.ErrNotAuthorized
```

Comment:
It might be worth also logging some information about the client - in particular, the client id could make it possible to look up if the client has history with us, also the labels typically make it possible what platform, client build, etc, the client is reporting.

(Another, big picture option, would be to ensure that the various authorizer methods log when returning false. This would have the advantage of being able to clearly log why the authorizer returned false.)

Reply:
Good point. I added more information (id, addr and labels). I wonder if "%v" is an appropriate formatting instruction in this case? In some cases we're going to be printing structs, but judging by the docs, that should be ok.

Reply:
I think id and addr both define a reasonable String method. I don't remember what happens to slices, but I think it isn't too crazy.

BTW, another place where we might count errors without logging them is in the handling of m.readOne's response - below, near line 293. The logic is meant to not log errors when the context is canceled, but we still still show the stats collector an error. I do not know how often that triggers.

Reply:
Checked the line 293. There's the following call next to the StatsCollector call:
```
log.Errorf("Streaming Connection to %v terminated with error: %v", m.info.Client.ID, err)
```

So I would presume that counted errors would correspond to logged errors in this case. Or am I missing something?

Reply:
JFYI: Will proceed with submitting this PR. Will address the cancelled context issue separately.